src/settings.cpp

```diff
@@ -766,18 +766,26 @@ bool Settings::setEntry(const std::string &name, const void *data,
 {
 	Settings *old_group = NULL;
 
+	// Return if setting name is invalid
 	if (!checkNameValid(name))
 		return false;
-	if (!set_group && !checkValueValid(*(const std::string *)data))
+
+	std::string value = *(const std::string *)data;
```

Comment:
Segfault if `set_group = false`.

Reply:
I'm not done yet, but thanks :)

Tbh, I don't understand what this line of code even does (it'd be great if someone could explain this to me), but I'm just assigning this to a variable early on, to make the code more readable.

Reply:
This line casts `data` (a void pointer) to a string pointer, then deferences it to get the string. You shouldn't do that if `set_group` is true because then `data` would point to a `Setttings` object, not a string.

Reply:
Oh ok. This makes sense now. Thanks!